platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -49,6 +49,42 @@ - (instancetype _Nonnull)initWithStyleURL:(nullable NSURL*)styleURL camera:(MGLM
 
 @end
 
+@interface MGLMapSnapshot()
+- (instancetype)initWithImage:(nullable MGLImage *)image scale:(CGFloat)scale pointForFn:(mbgl::MapSnapshotter::PointForFn)pointForFn;
+
+@property (nonatomic) CGFloat scale;
+@end
+
+@implementation MGLMapSnapshot {
+    mbgl::MapSnapshotter::PointForFn _pointForFn;
+}
+- (instancetype)initWithImage:(nullable MGLImage *)image scale:(CGFloat)scale pointForFn:(mbgl::MapSnapshotter::PointForFn)pointForFn;
+{
+    self = [super init];
+    if (self) {
+        _pointForFn = std::move(pointForFn);
+        _scale = scale;
+        _image = image;
+    }
+    return self;
+}
+
+#if TARGET_OS_IPHONE
+- (CGPoint)pointForCoordinate:(CLLocationCoordinate2D)coordinate;
+{
+    mbgl::ScreenCoordinate sc = _pointForFn(MGLLatLngFromLocationCoordinate2D(coordinate));
+    return CGPointMake(sc.x * self.scale, sc.y * self.scale);
+}
+
+#else
+- (NSPoint)pointForCoordinate:(CLLocationCoordinate2D)coordinate;
+{
+    mbgl::ScreenCoordinate sc = _pointForFn(MGLLatLngFromLocationCoordinate2D(coordinate));
+    return NSMakePoint(sc.x * self.scale, sc.y * self.scale);
+}
+#endif
```

Comment:
FYI, on macOS, `NSPoint` is a typedef for `CGPoint` and `NSMakePoint()` calls `CGPointMake()`, so conditional compilation isn’t strictly necessary in the implementation.